user_modules/shared_testcases/testCasesForPerformAudioPassThru.lua

```diff
@@ -0,0 +1,566 @@
+    {
+      initialPrompt =
+      {
+        {
+          text = string.rep("a", 500),
+          type = "TEXT",
+        }
+      },
+      audioPassThruDisplayText1 = string.rep("a", 500),
+      audioPassThruDisplayText2 = string.rep("a", 500),
+      samplingRate = "44KHZ",
+      maxDuration = 1000000,
+      bitsPerSample = "16_BIT",
+      audioType = "PCM",
+      muteAudio = true,
+      audioPassThruIcon =
+      { value = string.rep ("a", 251).. ".png",
+        imageType = "STATIC"
+      }
+    })
+
+  -- hmi expects TTS.Speak request
+  EXPECT_HMICALL("TTS.Speak",
+    {
+      speakType = "AUDIO_PASS_THRU",
+      ttsChunks = { { text = string.rep("a", 500), type = "TEXT" } },
+      appID = self.applications[applicationName]
+    })
+  :Do(function(_,data)
+      -- send notification to start TTS.Speak
+      self.hmiConnection:SendNotification("TTS.Started",{ })
+
+      -- HMI sends TTS.Speak SUCCESS
+      local function ttsSpeakResponse()
+        self.hmiConnection:SendResponse (data.id, data.method, "SUCCESS", {})
+
+        -- HMI sends TTS.Stop
+        self.hmiConnection:SendNotification("TTS.Stopped")
+      end
+
+      RUN_AFTER(ttsSpeakResponse, 1000)
+    end)
+
+  -- hmi expects UI.PerformAudioPassThru request
+  EXPECT_HMICALL("UI.PerformAudioPassThru",
+    {
+      appID = self.applications[applicationName],
+      audioPassThruDisplayTexts = {
+        {fieldName = "audioPassThruDisplayText1", fieldText = string.rep("a", 500)},
+        {fieldName = "audioPassThruDisplayText2", fieldText = string.rep("a", 500)},
+      },
+      maxDuration = 1000000,
+      muteAudio = true,
+      audioPassThruIcon = { imageType = "STATIC", value = string.rep ("a", 251).. ".png"}
+
+    })
+  :Do(function(_,data)
+      local function UIPerformAoudioResponce()
+        self.hmiConnection:SendResponse(data.id, "UI.PerformAudioPassThru", "SUCCESS", {})
+      end
+
+      RUN_AFTER(UIPerformAoudioResponce, 1500)
+    end)
+
+  if
+  self.appHMITypes["NAVIGATION"] == true or
+  self.appHMITypes["COMMUNICATION"] == true or
+  self.isMediaApplication == true then
+    --mobile side: expect OnHMIStatus notification
+    EXPECT_NOTIFICATION("OnHMIStatus",
+      {hmiLevel = "FULL", audioStreamingState = "ATTENUATED", systemContext = "MAIN"},
+      {hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN"})
+    :Times(2)
+  else
+    EXPECT_NOTIFICATION("OnHMIStatus")
+    :Times(0)
+  end
+
+  self.mobileSession:ExpectResponse(CorIdPerformAudioPassThruAppParVD, { success = true, resultCode = "SUCCESS",
+    })
+
+  commonTestCases:DelayedExp(1500)
+
+end
+
+function testCasesForPerformAudioPassThru:PerformAudioPassThru_AllParameters_Lower_SUCCESS(self)
+  local CorIdPerformAudioPassThruAppParVD= self.mobileSession:SendRPC("PerformAudioPassThru",
+    {
+      initialPrompt =
+      {
+        {
+          text = "",
+          type = "TEXT",
+        },
+
+      },
+      audioPassThruDisplayText1 = "1",
```

Comment:
@istoimenova Please check requirement [MOBILE_API]: SDL must set '1' as minlength of 'String' parameter by default in case minlength omitted in MOBILE_API